### PR TITLE
Fix for energy sampling in BetheHeithlerInteractor

### PR DESCRIPTION
**Pull request by paulromano**
Based on the discussion on slack, the probability for selecting the f1/g1 and f2/g2 functions in the BetheHeitlerInteractor is not correct. This PR uses the two-argument form of `BernoulliDistribution` with the values N1 and N2 from the Geant4 physics reference manual (so that the probability of selecting f1/g1 is N1/(N1 + N2)). I also put in a small fix in install-spack.sh.

### CODE REVIEW

File: scripts/dev/install-spack.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ info "This script is running from ${SCRIPT_DIR}"
 # Clone Spack and repositories
 ###############################################################################
 
-if [ -n "${SPACK_ROOT}" ]; then
+if [ ! -n "${SPACK_ROOT}" ]; then
```
Comment (sethrj) [inline scripts/dev/install-spack.sh:32]: Good catch -- I fixed this in the original one (which I clumsily copied, hence the older `File` name above):
```suggestion
if [ -z "${SPACK_ROOT}" ]; then
```

**Review by sethrj — COMMENTED** [review]
Did the change have any effect on the unit test? If not, it sounds like we need to increase coverage...

**Comment by paulromano** [comment]
@sethrj Yeah, it looks like the change didn't affect the unit test, so we're probably missing something in the parameter space that would affect this.

**Comment by sethrj** [comment]
Yeah with as complex a routine as this it might be difficult to cover the parameter space or meaningfully test it. We should plan on Wednesday on a validation strategy for new models as they come online.

**Review by sethrj — APPROVED** [review]

**Pull request merged**